test/RelayClient.test.ts

```diff
@@ -0,0 +1,222 @@
+      };
+      relayClient = new RelayClient();
+      provider = sandbox.createStubInstance(providers.JsonRpcProvider);
+      (relayClient as unknown as { _provider: providers.Provider })._provider =
+        provider;
+    });
+
+    afterEach(function () {
+      sandbox.restore();
+    });
+
+    it('should return the estimation applying the internal correction', async function () {
+      const estimateGas = BigNumber.from(10000);
+      provider.estimateGas.returns(Promise.resolve(estimateGas));
+      const internalGasCorrection = 5000;
+      const estimation = await relayClient.estimateInternalCallGas(
+        transactionDetails,
+        internalGasCorrection
+      );
+      const expectedEstimation = estimateGas.sub(internalGasCorrection);
+      expect(estimation.toString()).to.be.equal(expectedEstimation.toString());
+    });
+
+    it('should return the estimation without applying the internal correction', async function () {
+      const estimateGas = BigNumber.from(4000);
+      provider.estimateGas.returns(Promise.resolve(estimateGas));
+      const internalGasCorrection = 5000;
+      const estimation = await relayClient.estimateInternalCallGas(
+        transactionDetails,
+        internalGasCorrection
+      );
+      const expectedEstimation = estimateGas;
+      expect(estimation.toString()).to.be.equal(expectedEstimation.toString());
+    });
+
+    it('should return estimation without applying gas correction factor', async function () {
+      const stubApplyGasCorrection = sandbox
+        .stub(
+          relayClient as unknown as {
+            _applyGasCorrectionFactor: () => BigNumber;
+          },
+          '_applyGasCorrectionFactor'
+        )
+        .returns(BigNumber.from(7500));
+      const estimateGas = BigNumber.from(10000);
+      provider.estimateGas.returns(Promise.resolve(estimateGas));
+      const internalGasCorrection = 5000;
+      const estimation = await relayClient.estimateInternalCallGas(
+        transactionDetails,
+        internalGasCorrection,
+        false
+      );
+      const expectedEstimation = estimateGas.sub(internalGasCorrection);
+      expect(estimation.toString()).to.be.equal(expectedEstimation.toString());
+      expect(stubApplyGasCorrection.calledOnce).to.be.false;
+    });
+
+    it('should return estimation applying gas correction factor', async function () {
+      const expectedEstimation = BigNumber.from(7500);
+      const stubApplyGasCorrection = sandbox
+        .stub(
+          relayClient as unknown as {
+            _applyGasCorrectionFactor: () => BigNumber;
+          },
+          '_applyGasCorrectionFactor'
+        )
+        .returns(expectedEstimation);
+      const estimateGas = BigNumber.from(10000);
+      provider.estimateGas.returns(Promise.resolve(estimateGas));
+      const internalGasCorrection = 5000;
+      const estimation = await relayClient.estimateInternalCallGas(
+        transactionDetails,
+        internalGasCorrection
+      );
```

Comment:
It's a good opportunity to send true as third parameter and test something that hasn't been tested before.